infra/build/functions/trial_build.py

```diff
@@ -324,13 +355,17 @@ def _do_test_builds(args, test_image_suffix):
     for project, project_build_id in project_builds.items():
       build_ids[project].append(project_build_id)
 
-  return wait_on_builds(build_ids, credentials, build_lib.IMAGE_PROJECT)
+  return wait_on_builds(build_ids, credentials, build_lib.IMAGE_PROJECT,
+                        end_time)
 
 
 def trial_build_main(args=None, local_base_build=True):
   """Main function for trial_build. Pushes test images and then does test
   builds."""
   args = get_args(args)
+  timeout = int(os.environ.get('TIMEOUT', 0))
```

Comment:
Should we set a higher default value here? Otherwise would '0' as a default result in nonsensical behaviour?